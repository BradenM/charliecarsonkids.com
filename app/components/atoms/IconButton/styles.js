/**
 *
 * IconButton Styled Components
 *
 * Atom
 */

import styled from 'styled-components';
import Color from 'utils/color';

export const Wrapper = styled.div`
  padding: ${props => props.padding};
  transition: background-color 400ms ease-in-out;
  border-radius: 50%;

  div:first-child {
    padding: 2px;
  }

  &:hover {
    cursor: pointer;
    background-color: ${Color.nude.lighten(0.1).string()};
  }
`;
