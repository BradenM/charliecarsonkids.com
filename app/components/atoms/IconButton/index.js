/**
 *
 * IconButton
 *
 * Atom
 */

import Icon from 'atoms/Icon';
import PropTypes from 'prop-types';
import React from 'react';
import * as S from './styles';

const IconButton = ({ url, onClick, Component, ...props }) => (
  <S.Wrapper>
    <Component onClick={() => onClick(url)} {...props} />
  </S.Wrapper>
);

IconButton.defaultProps = {
  onClick: url => window.open(url),
  url: '_blank',
  Component: Icon,
};

IconButton.propTypes = {
  onClick: PropTypes.func,
  url: PropTypes.string,
  Component: PropTypes.func,
};

export default IconButton;
