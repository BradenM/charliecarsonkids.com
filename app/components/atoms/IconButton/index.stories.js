/**
 *
 * Stories for IconButton
 *
 * Atom
 */

import React from 'react';
import IconButton from '.';

export default {
  title: 'Design|Atoms/IconButton',
  component: IconButton,
};

export const facebookIconButton = () => (
  <IconButton name="facebook" url="https://www.facebook.com" />
);
