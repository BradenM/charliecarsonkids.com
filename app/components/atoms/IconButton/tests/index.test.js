/**
 *
 * Tests for IconButton
 *
 *
 */

import { render } from '@testing-library/react';
import React from 'react';
import IconButton from '../index';

describe('<IconButton />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<IconButton name="icon" />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<IconButton name="icon" />);
    expect(firstChild).toMatchSnapshot();
  });
});
