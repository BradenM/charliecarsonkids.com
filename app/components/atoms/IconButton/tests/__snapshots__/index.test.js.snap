// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton /> Should render and match the snapshot 1`] = `
<div
  class="styles__Wrapper-kccwaf-0 cqnkNJ"
>
  <div>
    <div>
      <div
        data-src="https://static.charliecarsonkids.com/assets/icons/icon-100.svg"
        src=""
      />
    </div>
  </div>
</div>
`;
